src/search/bucket/makeresult.go

```diff
@@ -3,9 +3,21 @@ package bucket
 import (
 	"github.com/hearchco/hearchco/src/search/engines"
 	"github.com/hearchco/hearchco/src/search/result"
+	"github.com/rs/zerolog/log"
 )
 
+// Returns nil on invalid data
 func MakeSEResult(urll string, title string, description string, searchEngineName engines.Name, sePage int, seOnPageRank int) *result.RetrievedResult {
```

Comment:
returned pointers are always heap allocated and there are a lot of RetrievedResults (Garbage Collector may go insane when we start having actual users)